pkg/cli/subcommands/workloadlist/command.go

```diff
@@ -0,0 +1,124 @@
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/pkg/api/util"
+	pkgconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+)
+
+// cliParams are the command-line arguments for this subcommand
+type cliParams struct {
+	GlobalParams
+
+	verboseList bool
+}
+
+type GlobalParams struct {
+	ConfFilePath string
+	ConfigName   string
+	LoggerName   string
+}
+
+// MakeCommand returns a `workload-list` command to be used by agent binaries.
+func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
+	cliParams := &cliParams{}
+
+	workloadListCommand := &cobra.Command{
+		Use:   "workload-list",
+		Short: "Print the workload content of a running agent",
+		Long:  ``,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			globalParams := globalParamsGetter()
+
+			cliParams.GlobalParams = globalParams
+
+			return fxutil.OneShot(workloadList,
+				fx.Supply(cliParams),
+				fx.Supply(core.BundleParams{
+					ConfigParams: config.NewAgentParamsWithoutSecrets(
+						globalParams.ConfFilePath,
+						config.WithConfigName(globalParams.ConfigName),
+					),
+					LogParams: log.LogForOneShot(globalParams.LoggerName, "off", true)}),
+				core.Bundle,
+			)
+		},
+	}
+
+	workloadListCommand.Flags().BoolVarP(&cliParams.verboseList, "verbose", "v", false, "print out a full dump of the workload store")
+
+	return workloadListCommand
+}
+
+func workloadList(log log.Component, config config.Component, cliParams *cliParams) error {
+	c := util.GetClient(false) // FIX: get certificates right then make this true
+
+	// Set session token
+	err := util.SetAuthToken()
+	if err != nil {
+		return err
+	}
+
+	url, err := workloadURL(cliParams.verboseList)
+	if err != nil {
+		return err
+	}
+
+	r, err := util.DoGet(c, url, util.LeaveConnectionOpen)
+	if err != nil {
+		if r != nil && string(r) != "" {
+			fmt.Fprintf(color.Output, "The agent ran into an error while getting the workload store information: %s\n", string(r))
+		} else {
+			fmt.Fprintf(color.Output, "Failed to query the agent (running?): %s\n", err)
+		}
+	}
+
+	workload := workloadmeta.WorkloadDumpResponse{}
+	err = json.Unmarshal(r, &workload)
+	if err != nil {
+		return err
+	}
+
+	workload.Write(color.Output)
+
+	return nil
+}
+
+func workloadURL(verbose bool) (string, error) {
+	ipcAddress, err := pkgconfig.GetIPCAddress()
+	if err != nil {
+		return "", err
+	}
+
+	var prefix string
+	if flavor.GetFlavor() == flavor.ClusterAgent {
+		prefix = fmt.Sprintf("https://%v:%v/workload-list", ipcAddress, pkgconfig.Datadog.GetInt("cluster_agent.cmd_port"))
+	} else {
+		prefix = fmt.Sprintf("https://%v:%v/agent/workload-list", ipcAddress, pkgconfig.Datadog.GetInt("cmd_port"))
+	}
+
+	if verbose {
+		return prefix + "/verbose", nil
```

Comment:
Following my suggestion:

```suggestion
		return prefix + "?verbose=true", nil
```